src/coreComponents/dataRepository/BufferOps_inline.hpp

```diff
@@ -1241,6 +1241,7 @@ Pack( buffer_unit_type * & buffer,
       arrayView1d< globalIndex const > const & relatedObjectLocalToGlobalMap )
 {
   localIndex sizeOfPackedChars=0;
+  array1d< globalIndex > junk;
```

Comment:
Good question! The change here is simply moving the declaration of the arrays outside the for loop since it is not needed to be declared repeatedly. Our profiling results showed that by simply applying these changes (at the five pack / unpack functions) could provide performance improvements at initialization phase by 10% to 20%, especially in the cases when indices.size() is large. 

I have discussed this with Sergey a bit and you are right that the "junk" array is probably not needed at all. I agree that there would be better approach, but might need a lot more changes (resulting modifications of interfaces at difference functions). As it serves the purpose (for initialization performance improvement) at this stage, we commit the changes as it and for sure we could discuss and consider better solutions later.